compiler/dyno/lib/uast/Builder.cpp

```diff
@@ -280,12 +289,118 @@ void Builder::doAssignIDs(AstNode* ast, UniqueString symbolPath, int& i,
   if (search != notedLocations_.end()) {
     assert(!search->second.isEmpty());
     idToLocation_[ast->id()] = search->second;
+    if (ieNode) {
+      checkConfigPreviouslyUsed(ast, configNameUsed);
+    }
   } else {
     assert(false && "Location for all ast should be set by noteLocation");
   }
 }
 
-AstList Builder::flattenTopLevelBlocks(AstList lst) {
+  void Builder::checkConfigPreviouslyUsed(const AstNode* ast, std::string& configNameUsed) {
+    parsing::useConfigParam(context(), configNameUsed, ast->id());
+    auto usedId = parsing::nameToConfigParamId(context(), configNameUsed);
+
+    if (usedId != ast->id()) {
```

Comment:
Does this work? I would expect `useConfigParam` to replace whatever ID we had already and so this block would never run.

Reply:
Your expectation aligns with mine. What I experienced though is that because the revision# isn't incremented, when we re-issue this query, we get the cached value back instead of updating.